### PR TITLE
Contribution guidelines, update `.gdextension` configurations

### LINKED ISSUE #208: Error loading extension: res//DodgeTheCreeps.gdextension
I've tried to run DodgeTheCreeps demo on M1 Macbook Pro. The project built and the game started, but nothing happened after clicking the "START" button inside the game and following errors appeared:

```
Can't open dynamic library: /Users/mareksubocz/it/gdext/examples/dodge-the-creeps/godot/../../../target/debug/libdodge_the_creeps.dylib, error: dlopen(/Applications/Godot.app/Contents/MacOS/../Frameworks/libdodge_the_creeps.dylib, 0x0002): tried: '/Applications/Godot.app/Contents/MacOS/../Frameworks/libdodge_the_creeps.dylib' (no such file).
  core/extension/gdextension.cpp:400 - GDExtension dynamic library not found: /Users/mareksubocz/it/gdext/examples/dodge-the-creeps/godot/../../../target/debug/libdodge_the_creeps.dylib
  Failed loading resource: res://DodgeTheCreeps.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
  Can't open dynamic library: /Users/mareksubocz/it/gdext/examples/dodge-the-creeps/godot/../../../target/debug/libdodge_the_creeps.dylib, error: dlopen(/Applications/Godot.app/Contents/MacOS/../Frameworks/libdodge_the_creeps.dylib, 0x0002): tried: '/Applications/Godot.app/Contents/MacOS/../Frameworks/libdodge_the_creeps.dylib' (no such file).
  core/extension/gdextension.cpp:400 - GDExtension dynamic library not found: /Users/mareksubocz/it/gdext/examples/dodge-the-creeps/godot/../../../target/debug/libdodge_the_creeps.dylib
  Failed loading resource: res://DodgeTheCreeps.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
```

The steps I took to set the project up were:
```
brew install godot
git clone https://github.com/godot-rust/gdext && cd gdext
export GODOT4_BIN=/Applications/Godot.app/Contents/MacOS/Godot
export LLVM_PATH=/Library/Developer/CommandLineTools/usr/bin
cargo build # successfully
cd examples/dodge-the-creeps
cargo build # successfully
godot # to run the game
```

Is there something I overlooked?

### PR DESCRIPTION
**Pull request by Bromeon**
Elaborates the contribution process, documents `#[itest]` and unifies the `.gdextension` library entries.

Fixes #208.

### CODE REVIEW

**Comment by Bromeon** [comment]
bors try

File: ReadMe.md [hunk 1]
```diff
@@ -80,7 +79,7 @@ To register the GDExtension library with Godot, you need to create two files rel
 We highly recommend to have a look at a working example in the `examples/dodge-the-creeps` directory.
 This integrates a small game with Godot and has all the necessary steps set up.
 
-API documentation can be generated locally using `cargo doc -p godot --no-deps --open`.
+API documentation can be generated locally using `./check.sh dok` (or `./check doc` to refresh an opened page).
```
Comment (gg-yb) [inline ReadMe.md:82]:
```suggestion
API documentation can be generated locally using `./check.sh dok` (or `./check.sh doc` to refresh an opened page).
```

Might not be the place for discussing it, but just mentioning "dok" as a key is prone to being considered a typo IMO. Maybe we should make it clear (here?) that "dok" (sic!) is meant

Comment (Bromeon) [reply ReadMe.md:82]: Thanks! Yeah, originally there was just `doc` and I found myself always wanting to open the browser. Maybe swapping the sentence in the parentheses with the one outside could clarify this.

**Pull request merged**